scripts/test-bleichenbacher-workaround.py

```diff
@@ -1123,12 +1123,10 @@ def main():
     # make sure that sanity test is run first and last
     # to verify that server was running and kept running through
     sanity_tests = [('sanity', conversations['sanity'])]
-    ordered_tests = chain(sanity_tests,
-                          islice(filter(lambda x: x[0] != 'sanity',
-                                        conversations.items()), num_limit),
-                          sanity_tests)
+    regular_tests = [(k, v) for k, v in conversations.items() if k != 'sanity']
+    sampled_tests = sample(regular_tests, min(num_limit, len(regular_tests)))
 
-    for c_name, c_test in ordered_tests:
+    for c_name, c_test in sanity_tests + sampled_tests + sanity_tests:
```

Comment:
while this will work, it will create multiple temporary lists, impacting performance - that's why old code used `chain()`